src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -288,20 +288,6 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
                 .SetBasePath(_hostingEnvironment.ContentRootPath)
                 .AddInMemoryCollection(_config.AsEnumerable());
 
-            foreach (var configureAppConfiguration in _configureAppConfigurationBuilderDelegates)
-            {
-                configureAppConfiguration(_context, builder);
-            }
-
-            var configuration = builder.Build();
-            services.AddSingleton<IConfiguration>(configuration);
-            _context.Configuration = configuration;
-
-            // The configured ILoggerFactory is added as a singleton here. AddLogging below will not add an additional one.
-            var loggerFactory = _createLoggerFactoryDelegate?.Invoke(_context) ?? new LoggerFactory(configuration.GetSection("Logging"));
-            services.AddSingleton(loggerFactory);
-            _context.LoggerFactory = loggerFactory;
-
             if (!_options.PreventHostingStartup)
```

Comment:
Keep going, all the way to line 275. E.g. the _hostingEnvironment construction feeds directly into the ServiceCollection construction, but now you've split them.